### ISSUE TITLE
MacOS stage of Travis CI is broken

### ISSUE THREAD

**Issue by clinssen** [nest/nest-simulator#1822]
This looks like an issue with a Python 3.8 vs. 3.9 version conflict. Possibly, this is due to the default Python version in the Travis CI MacOS image having been upgraded to 3.9.

**Comment by clinssen** [comment]
And, on top of this, some new news from Travis CI: https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing

If I'm reading this right, this means we have to move to a paid (subscription) plan to continue building on MacOS, or apply for OSS credits.


cc @lekshmideepu

**Comment by heplesser** [comment]
Looking at the build log in #1826, the first thing I noticed is an failure to build certain code in connection with installation of matplotlib (lines 777-808 in https://travis-ci.org/github/nest/nest-simulator/jobs/741134305) ending in 
```
 ImportError: dlopen(/usr/local/lib/python3.9/site-packages/numpy/core/_multiarray_umath.cpython-39-darwin.so, 2): Symbol not found: ___addtf3
809      Referenced from: /usr/local/opt/gcc/lib/gcc/10/libquadmath.0.dylib
810      Expected in: /usr/lib/libSystem.B.dylib
811     in /usr/local/opt/gcc/lib/gcc/10/libquadmath.0.dylib
```

I found some older posts discussion missing `__addtf3`([one](https://github.com/ContinuumIO/anaconda-issues/issues/3455), [another](https://github.com/matplotlib/matplotlib/issues/12578)). 

I also noted that a lot of time is spent on building NumPy, which is installed via pip. I wonder if we could avoid this by installing binaries from suitable archives. This would have the advantage that at least all package versions with pertaining libraries should be consistent. Alternatives would be Miniconda, Homebrew and MacPorts.

Homebrew does not provide Matplotlib, so it cannot be the one solution solving (almost) all. MacPorts seems to provide almost all we need, although at present MPI4Py only up to Python 3.8. But maybe we do not have to use 3.9 on Travis already?
